protocol/requests/delete_community_member_messages.go

```diff
@@ -0,0 +1,54 @@
+package requests
+
+import (
+	"errors"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+var ErrDeleteCommunityMemberMessagesInvalidCommunityID = errors.New("delete-community-member-messages: invalid community id")
+var ErrDeleteCommunityMemberMessagesInvalidMemberID = errors.New("delete-community-member-messages: invalid member id")
+var ErrDeleteCommunityMemberMessagesInvalidData = errors.New("delete-community-member-messages: invalid data")
+var ErrDeleteCommunityMemberMessagesInvalidDeleteAll = errors.New("delete-community-member-messages: invalid delete all setup")
+var ErrDeleteCommunityMemberMessagesInvalidDeleteMessagesByID = errors.New("delete-community-member-messages: invalid delete messages by ID setups")
+var ErrDeleteCommunityMemberMessagesInvalidMsgID = errors.New("delete-community-member-messages: invalid messages Id")
+var ErrDeleteCommunityMemberMessagesInvalidMsgChatID = errors.New("delete-community-member-messages: invalid messages chatId")
+
+type DeleteCommunityMemberMessages struct {
+	CommunityID  types.HexBytes                           `json:"communityId"`
+	MemberPubKey string                                   `json:"memberPubKey"`
+	Messages     []*protobuf.DeleteCommunityMemberMessage `json:"messages"`
+	DeleteAll    bool                                     `json:"deleteAll"`
+}
+
+func (d *DeleteCommunityMemberMessages) Validate() error {
+	if len(d.CommunityID) == 0 {
+		return ErrDeleteCommunityMemberMessagesInvalidCommunityID
+	}
+
+	if len(d.MemberPubKey) == 0 {
+		return ErrDeleteCommunityMemberMessagesInvalidMemberID
+	}
+
+	if d.Messages != nil && len(d.Messages) > 0 && d.DeleteAll {
+		return ErrDeleteCommunityMemberMessagesInvalidDeleteAll
+	}
+
+	if (d.Messages == nil || d.Messages != nil && len(d.Messages) == 0) && !d.DeleteAll {
```

Comment:
This first part if confusing, because there is an OR and an AND in the same parentheses. I guess you meant
`(d.Messages == nil || (d.Messages != nil && len(d.Messages) == 0))` ?

Reply:
OR will split validation into 2 parts and part 2 will be executed only if validation 1 failed
validation 1 - `d.Messages == nil`
validation 2 - `d.Messages != nil && len(d.Messages) == 0)` 
so it is working correctly, but I'll change it for more better readiness